lib/screens/editor/versions.dart

```diff
@@ -0,0 +1,398 @@
+import 'dart:collection';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:every_door/private.dart';
+import 'package:every_door/providers/osm_api.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:http/http.dart' as http;
+import 'package:intl/intl.dart';
+import 'package:url_launcher/url_launcher.dart';
+
+class TagChange {
+  final String? oldValue;
+  final String? newValue;
+  bool get noChange => oldValue == newValue;
+  bool get removed => newValue == null;
+  bool get added => oldValue == null;
+  bool get updated => !added && !removed && oldValue != newValue;
+
+  const TagChange(this.oldValue, this.newValue);
+
+  Color? getColor() {
+    if (noChange) return null;
+    if (added) return Colors.green.shade100;
+    if (updated) return Colors.amber.shade200;
+    if (removed) return Colors.red.shade100;
+
+    // should never be reached
+    return null;
+  }
+
+  @override
+  String toString() {
+    return "was: $oldValue, now: $newValue";
```

Comment:
Could you please format toStrings as `ClassName(...)`? E.g. here `TagChange($oldValue, $newValue)` might look better. Such format is easier to read when there are objects nested.

Reply:
Fixed in 2e563be166f59791df1185aa2e656523515cf28e